TradeDis/frontend
# feat(posts): create post frontend

**What was done?**
Story: TRAD-2 "As a user, I would like to be able to create an inquiry posting on the main feed so others can see it"
Task: TRAD-69 UI - Post Creation Page
- Simple UI screen to add new posts
- Backend connection to `/api/v1/posts` configured with axios

**Areas which need special reviewing attention**
- Specifically the `useState` React hook section. This was my first time using React hooks so @alanwu4321 or @j584lee98 if you guys could take a look that would be great.

**Future Improvement**
- Better styling (centering elements, etc)
- Better response/error handling on backend call
- Required fields and error handling on frontend
- Populate post object fully (with user credentials, etc)

**Testing**
Testing that creating a post works
1. In `NewPostScreen.tsx` on line 44, replace `localhost` with your IP address
2. Ensure the backend server is running (`yarn dev` in backend project) as well as expo (`expo start`)
3. Fill in the fields shown on the screen and press on the "Post!" button
4. A success message should show under the button
5. Navigate to `http://localhost:3000/api/v1/posts` and your newly added post should be there

This is how the screen currently looks. Improvement will definitely be made 😂
![IMG_1617](https://user-images.githubusercontent.com/48066840/97126725-07a27380-170e-11eb-97a7-816db028c7db.PNG)
